### DISCUSSION TITLE
Raw vs Curses

### DISCUSSION THREAD

**Discussion by mvanderkamp** [inducer/pudb#592] [Q&A]
I was curious about issue #48 and poking around in the code when I noticed the following on [pudb/debugger.py:640](https://github.com/inducer/pudb/blob/main/pudb/debugger.py#L640):

```
if curses is not None:
    class ThreadsafeCursesScreen(ThreadsafeScreenMixin, RawScreen):
        pass
```

Should this be subclassing CursesScreen instead? What difference in behaviour should we expect from the two base classes?

**Comment by inducer** [comment]
Yeah, that looks like a copy-paste fail. Could you do a PR?

**Reply by mvanderkamp** [reply]
Sure. Should I expect that my normal terminal/shell would handle a curses display properly, or is there a particular kind of terminal/shell I should test with?

**Reply by inducer** [reply]
My understanding of the curses display is that it *might* do better than the more efficient raw display on old-timey terminals that don't conform to the "vaguely xterm-ish" set of ANSI sequences that everything seems to understand these days. But curses should definitely work fine there, too.
